src/broker/broker.c

```diff
@@ -1528,13 +1529,31 @@ static void module_cb (module_t *p, void *arg)
             (void)broker_response_sendmsg (ctx, msg);
             break;
         case FLUX_MSGTYPE_REQUEST:
-            broker_request_sendmsg (ctx, msg);
-            if (flux_msg_get_route_count (msg) == 1
-                && module_disconnect_arm (p, msg, disconnect_send_cb, ctx) < 0) {
-                    flux_log_error (ctx->h,
-                                    "%s: module_disconnect_arm",
-                                    module_get_name (p));
+            count = flux_msg_get_route_count (msg);
+            /* Requests originated by the broker module will have a route
+             * count of 1.  Ensure that, when the module is unloaded, a
+             * disconnect message is sent to all services used by broker module.
+             */
+            if (count == 1) {
+                if (module_disconnect_arm (p, msg, disconnect_send_cb, ctx) < 0)
+                    flux_log_error (ctx->h, "error arming module disconnect");
             }
+            /* Requests sent by the module on behalf of _its_ peers, e.g.
+             * connector-local module with connected clients, will have a
+             * route count greater than one here.  If this broker is not yet
+             * "online" (entered INIT state), politely rebuff these requests.
+             * Possible scenario for this message: user submitting a job on
+             * a login node before cluster reboot is complete.
+             */
+            else if (count > 1 && !ctx->online) {
+                const char *errmsg = "Upstream Flux broker is not yet online."
+                                     " Try again later.";
```

Comment:
Would it make the error more general to just say `Upstream broker is offline`? Or does this really only apply at startup when brokers have "not yet" been online?

Reply:
Offline is probably better, even though we say we don't upstream to go down without shutting down downstream.  Soon that will have to change.